### DISCUSSION TITLE
Current impl of locale settings saving is flawed

### DISCUSSION THREAD

**Discussion by PaiJi** [Dokploy/dokploy#718] [General]
### Background info

The i18n solution we used in the project will prepare translation in the SSR stage, so the language setting must be available on the first request from a cookie or something else.

But:
- On Chrome, cookies expire after 400 days anyway.
- On Safari, Cookies expire after 7 days of the user not visiting the page.
- On Brave, Cookies will always expire after 7 days.

more info come from https://www.cookiestatus.com/

### Goal

We certainly don't want users to experience their language preferences expiring after a period of time. therefore, I hope the new language preference settings will meet the following requirements.

1. Language preferences should not be forgotten, at least not while the login session is still active.
2. Language preferences remain synchronized after switching devices.
3. It's best not to hard-reload the entire page when switching languages to avoid losing user input.

###  Possible impl

1. Storage language preferences into database or where userInfo storage.
2. Save language preferences into auth cookie.
3. Get language preferences from auth cookie.

**Comment by PaiJi** [comment]
Any idea? @Siumauricio

**Reply by Siumauricio** [reply]
@PaiJi I agree with the application, I am open to receive a PR
